src/wp-includes/script-loader.php

```diff
@@ -3684,33 +3692,50 @@ public function get_span(): WP_HTML_Span {
 			};
 
 			// Loop over STYLE tags.
-			while ( $processor->next_tag( array( 'tag_name' => 'STYLE' ) ) ) {
-				// Skip to the next if this is not the inline style for the wp-block-library stylesheet (which contains the placeholder).
-				if ( 'wp-block-library-inline-css' !== $processor->get_attribute( 'id' ) ) {
-					continue;
-				}
+			while ( $processor->next_tag( array( 'tag_closers' => 'visit' ) ) ) {
+
+				// We've encountered the inline style for the 'wp-block-library' stylesheet which probably has the placeholder comment.
+				if (
+					! $processor->is_tag_closer() &&
```

Comment:
this is redundant here because `STYLE` is a special atomic element which behaves like a void element. if the tag name is `STYLE` it will _always_ return `! $processor->is_tag_closer()`